adafruit_tinylora/adafruit_tinylora.py

```diff
@@ -0,0 +1,405 @@
+
+    def set_datarate(self, datarate):
+        """Sets the RFM Datarate
+        :param datarate: Bandwidth and Frequency Plan
+        """
+        if datarate == 'SF7BW125':
+            self._sf = 0x74
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF7BW250':
+            self._sf = 0x74
+            self._bw = 0x82
+            self._modemcfg = 0x04
+        elif datarate == 'SF8BW125':
+            self._sf = 0x84
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF9BW125':
+            self._sf = 0x94
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF10BW125':
+            self._sf = 0xA4
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF11BW125':
+            self._sf = 0xB4
+            self._bw = 0x72
+            self._modemcfg = 0x0C
+        elif datarate == 'SF12BW125':
+            self._sf = 0xC4
+            self._bw = 0x72
+            self._modemcfg = 0x0C
+        else:
+            raise TypeError("Invalid Datarate.")
+
+    def set_channel(self, channel):
+        """Returns the RFM Channel (if single-channel)
+        :param int channel: Transmit Channel (0 through 7).
+        """
+        if self._channel is not None:
+            if channel == 0:
+                self._rfm_lsb = self._frequencies[0][2]
+                self._rfm_mid = self._frequencies[0][1]
+                self._rfm_msb = self._frequencies[0][0]
+            elif channel == 1:
+                self._rfm_lsb = self._frequencies[1][2]
+                self._rfm_mid = self._frequencies[1][1]
+                self._rfm_msb = self._frequencies[1][0]
+            elif channel == 2:
+                self._rfm_lsb = self._frequencies[2][2]
+                self._rfm_mid = self._frequencies[2][1]
+                self._rfm_msb = self._frequencies[2][0]
+            elif channel == 3:
+                self._rfm_lsb = self._frequencies[3][2]
+                self._rfm_mid = self._frequencies[3][1]
+                self._rfm_msb = self._frequencies[3][0]
+            elif channel == 4:
+                self._rfm_lsb = self._frequencies[4][2]
+                self._rfm_mid = self._frequencies[4][1]
+                self._rfm_msb = self._frequencies[4][0]
+            elif channel == 5:
+                self._rfm_lsb = self._frequencies[5][2]
+                self._rfm_mid = self._frequencies[5][1]
+                self._rfm_msb = self._frequencies[5][0]
+            elif channel == 6:
+                self._rfm_lsb = self._frequencies[6][2]
+                self._rfm_mid = self._frequencies[6][1]
+                self._rfm_msb = self._frequencies[6][0]
+            elif channel == 7:
+                self._rfm_lsb = self._frequencies[7][2]
+                self._rfm_mid = self._frequencies[7][1]
+                self._rfm_msb = self._frequencies[7][0]
```

Comment:
If you setup your `TTN_FREQS` as nested tuples, this entire thing could become one line:
```python
self._rfm_msb, self._rfm_mid, self._rfm_lsb = self._frequencies[channel]
```
**NOTE** the ordering for the tuple unpacking

Reply:
Adding, this is a much better solution and saves a bunch of space. Changing `TTN_FREQS` within the support files to reflect the change to nested tuples.